binary_transparency/firmware/cmd/ft_monitor/main.go

```diff
@@ -116,13 +121,27 @@ func main() {
 				glog.Warningf("Unable to GetFirmwareImage for Firmware with Hash 0x%x , reason %q", meta.FirmwareImageSHA512, err)
 				continue
 			}
+
 			// Verify Image Hash from log Manifest matches the actual image hash
 			h := sha512.Sum512(image)
 			if !bytes.Equal(h[:], meta.FirmwareImageSHA512) {
 				glog.Warningf("downloaded image does not match SHA512 in metadata (%x != %x)", h[:], meta.FirmwareImageSHA512)
 				continue
 			}
 			glog.V(1).Infof("Image Hash Verified for image at leaf index %d", manifest.LeafIndex)
+
+
+			//Search for specific keywords inside firmware image
+			for _, keyword := range strings.Fields(ftKeyWords) {
```

Comment:
just a thought - you could do a bunch of this work just once at line 68:

```go
var ftKeywords []*regex.Regexp
for _, k := range strings.Fields(*keyWord) {
  ftKeywords = append(ftKeywords, regex.MustCompile(k))
}
```

then down here becomes just

```go
for _, m := range ftKeywords {
  if m.Match(image) {
     // ARGH!
  }
}
```

It might also be worth considering explicitly calling out the flag as being a regex, and potentially just letting the user specify things like `\b(malware|virus|malicious|...)\b` in the flag directly - the regex compiler will be able to make the searches sublinear then.

Reply:
Agree on your 1st Comment, will consider better aligning code, the way you indicated. 
However, I request, we just keep the input for user to be simple a string with bunch of keywords, which user wants to search for, **like below.** He should not care to construct something special like regular expression!

"XYZ PQR LMQ"

Reply:
Sure thing

Reply:
All comments completed and done. Please review and merge!